### ISSUE TITLE
Feature request: sidebar waterfall menu

### ISSUE THREAD

**Issue by GFdevelop** [geany/geany#1718]
Instead of showing the tabs it would be more convenient if a mouseover (without clicking) opens a cascade menu with the symbols, documents, plugin tabs, etc...
It would be a much faster and practical solution with just one click opens the desired tab.

**Comment by codebrainz** [comment]
Would it be possible to explain it terms of GTK+ widgets/terminology (or with a mock-up)?

**Comment by GFdevelop** [comment]
I don't know GTK but you can see my idea here https://www.w3schools.com/howto/howto_css_dropdown.asp move mouse over "Hover Me"

**Comment by codebrainz** [comment]
It's a bit unusual (and some would consider annoying) to have stuff randomly popup when you move your mouse around. With newer GTK+ this could be implemented with GtkMenuButton and/or GtkPopover, but I'm not sure if most people would like it.

**Comment by GFdevelop** [comment]
Is possible to make a settings option? The user can select if open menu with a click on it or with only mouse over it...

**Comment by elextr** [comment]
I think plugins add tabs directly to the notebook, so it would need some wrangling to get them into the new UI artifact.

**Comment by elextr** [comment]
Could be a plugin listening to the add/remove page signals.

**Comment by GFdevelop** [comment]
I do not know if I explained myself well, the menu should contain only the titles of the tabs, and clicking on an item such as "Symbols" opens the Symbols tab.

**Comment by codebrainz** [comment]
What do you mean by "menu" (ie. a button, a drop-down box, a main-menu like at the top of Geany), and where would it be placed within the UI?

**Comment by kugel-** [comment]
Correct me if I'm wrong but I think @GFdevelop asks for making the sidebar a GtkStack, with the stack switcher being a drop-down menu (possibly opening on mouse-over instead if normal click).

**Comment by codebrainz** [comment]
@kugel- yeah but where would the stack-switching drop-down menu live in the GUI?

**Comment by codebrainz** [comment]
Like this?

![sidebar_menu](https://user-images.githubusercontent.com/181177/34034702-616fb2a0-e134-11e7-8f56-41f02e78cf64.png)

**Comment by kugel-** [comment]
The drop-down would contain the label, so in place of the tab.

**Comment by GFdevelop** [comment]
What is GTKStack?

This is right 
> possibly opening on mouse-over instead if normal click

and this
> The drop-down would contain the label, so in place of the tab.

**Comment by GFdevelop** [comment]
![export](https://user-images.githubusercontent.com/17930457/34047415-7e30b55c-e1b0-11e7-859e-043a1a10dfa9.png)
![export 1](https://user-images.githubusercontent.com/17930457/34047424-81936582-e1b0-11e7-8d9e-ec1c23e9b31f.png)
![export 2](https://user-images.githubusercontent.com/17930457/34047428-843152c2-e1b0-11e7-9d1c-7d10866c7090.png)

**Comment by b4n** [comment]
We could have a [combobox](https://developer.gnome.org/gtk3/stable/GtkComboBox.html) dropdown and [hide the notebook tabs](https://developer.gnome.org/gtk3/stable/GtkNotebook.html#GtkNotebook--show-tabs).  This should likely be an option, and the popup-on-hover seems highly questionable to me as it's not common in UIs I know, and it can get super annoying.  I'm rather used to the click-to-open-and-release-on-the-element for a single-click menu usage.

**Comment by codebrainz** [comment]
> This should likely be an option, and the popup-on-hover seems highly questionable to me as it's not common in UIs I know, and it can get super annoying.

+1, but I think this is the whole point of the PR.

> We could have a combobox dropdown and hide the notebook tabs.

Seems kind of unusual. At that point might as well just put a right-click tab menu similar to what the editor notebook has.

**Comment by b4n** [comment]
> At that point might as well just put a right-click tab menu similar to what the editor notebook has.

Why not, and I think it's super easy just enabling [`GtkNotebook:enable-popup`](https://developer.gnome.org/gtk3/stable/GtkNotebook.html#GtkNotebook--enable-popup).

**Comment by elextr** [comment]
How do you have a right click tab menu if the tabs are hidden?

**Comment by codebrainz** [comment]
> Why not, and I think it's super easy

I wouldn't mind, but I don't think it's really what @GFdevelop is after.

> How do you have a right click tab menu if the tabs are hidden?

The tabs wouldn't be hidden.

**Comment by elextr** [comment]
> The tabs wouldn't be hidden.

As you say above, AFAIU thats not what the OP was after.

**Comment by eht16** [comment]
Since the popup menu has been long time ago, I think we can close this.

Otherwise, feel free to re-open.
